clients/filesystem-hadoop/build.gradle.kts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+plugins {
+  `maven-publish`
+  id("java")
+}
+
+dependencies {
+  compileOnly(libs.hadoop3.common)
+  implementation(project(":clients:client-java-runtime", configuration = "shadow"))
+
+  testImplementation(libs.hadoop3.common)
+  testImplementation(libs.junit.jupiter.api)
+  testImplementation(libs.junit.jupiter.params)
+  testImplementation(libs.mockito.core)
+  testImplementation(libs.mockserver.netty) {
+    exclude("com.google.guava", "guava")
+  }
+  testRuntimeOnly(libs.junit.jupiter.engine)
+}
+
+tasks.named("generateMetadataFileForMavenJavaPublication") {
+  dependsOn(":clients:filesystem-hadoop:copyDepends")
+}
```

Comment:
Refer to the `trino-connector` implementation here, and copy the local dependencies to `libs` before publishing maven. If we only publishing the `filesystem-hadoop` jar to maven, this may not be necessary.